lib/Reconciler.lua

```diff
@@ -44,14 +44,47 @@ local Reconciler = {}
 
 Reconciler._singleEventManager = SingleEventManager.new()
 
+--[[
+	Is this element backed by a Roblox instance directly?
+]]
+local function isPrimitiveElement(element)
+	if type(element) ~= "table" then
+		return false
+	end
+
+	return type(element.type) == "string"
+end
+
+--[[
+	Is this element defined by a pure function?
+]]
+local function isFunctionalElement(element)
+	if type(element) ~= "table" then
+		return false
+	end
+
+	return type(element.type) == "function"
+end
+
+--[[
+	Is this element defined by a component class?
+]]
+local function isStatefulElement(element)
+	if type(element) ~= "table" then
+		return false
+	end
+
+	return type(element.type) == "table"
+end
```

Comment:
Ah, can we stick these up next to the `isPortal` function? It used to live in `Core` as well, but no longer does.